src/interfaces.ts

```diff
@@ -316,3 +316,46 @@ export interface ISdkOptions {
 export interface IFeatureFlags {
     [key: string]: number | boolean | string;
 }
+
+export type GuidType = 'guid';
+export type RGBType = 'rgb';
+
+export interface IRGBColor {
+    r: number;
+    g: number;
+    b: number;
+}
+
+export interface IGuidColorItem {
+    type: GuidType;
+    value: string;
+}
+
+export interface IRGBColorItem {
+    type: RGBType;
+    value: IRGBColor;
+}
+
+export type IColorItem = IGuidColorItem | IRGBColorItem;
```

Comment:
```
export type ColorType = 'GUID' | 'RGB';

export interface IColorItem {
    type: ColorType;
}

export interface IGuidColorItem extends IColorItem {
    value: string;
}

export interface IRGBColorItem extends IColorItem {
    value: IRGBColor;
}

const item1: IRGBColorItem = {
    type: 'GUID',
    value: {
        b: 2,
        g: 1,
        r: 0
    }
};

const item2: IGuidColorItem = {
    type: 'GUID',
    value: 'color'
};

const items: IColorItem[] = [item1, item2];

function isGuidColor(item: IColorItem): item is IGuidColorItem {
    return item.type === 'GUID';
}

items.forEach((item) => {
   if (isGuidColor(item)) {
       item.value = 'newColor';
   }
});

export interface IRGBColor {
    r: number;
    g: number;
    b: number;
}
```
I would say this will be better approach :)

Reply:
Your solution allows for objects like `item1` in your example (i.e. GUID item with RGB values which seems nonsensical) which the current solution does not. I believe discriminated union types are a better fit here as I think `item1` really should not be valid.

Reply:
my bad ;)